test/functional/fixtures/api/es-next/selector/testcafe-fixtures/selector-test.js

```diff
@@ -525,3 +525,246 @@ test('Snapshot property shorthand - selector error', async () => {
 test('Snapshot shorthand method - selector error', async () => {
     await Selector(() => [].someUndefMethod()).hasClass('yo');
 });
+
+test('Selector "nth()" method', async () => {
+    // String selector
+    const getSecondEl = Selector('.idxEl').nth(1);
+
+    let el = await getSecondEl();
+
+    expect(el.id).eql('el2');
+
+    // Function selector
+    const getThirdEl = Selector(() => document.querySelectorAll('.idxEl')).nth(2);
+
+    el = await getThirdEl();
+
+    expect(el.id).eql('el3');
+
+    // If single node is returned index should be always 0
```

Comment:
It's clear. If you do `Selector('.i-m-the-only-element').nth(3)`, we should return null.